core/snmp/api/src/main/java/org/opennms/netmgt/snmp/SnmpV3User.java

```diff
@@ -106,4 +108,39 @@ public void setPrivProtocol(String privacyProtocol) {
         this.privProtocol = privacyProtocol;
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hash(authPassPhrase, authProtocol, engineId, privPassPhrase, privProtocol);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (o == null) {
+            return false;
+        }
+        if (o == this) {
+            return true;
+        }
+        if (this.getClass() != o.getClass()) {
```

Comment:
Can we use `instanceof` instead?
Usually getClass() should not be a problem, but can cause issues when a class is proxied or when subclassed.